project.clj

```diff
@@ -247,11 +246,32 @@
   ;; "test"].  See the :testutils profile below.
   :classifiers  {:test :testutils}
 
-  :profiles {:dev {:resource-paths ["test-resources"],
-                   :dependencies ~pdb-dev-deps
-                   :injections [(do
-                                  (require 'schema.core)
-                                  (schema.core/set-fn-validation! true))]}
+  :profiles {:defaults {:resource-paths ["test-resources"]
+                        :dependencies ~pdb-dev-deps
+                        :injections [(do
+                                       (require 'schema.core)
+                                       (schema.core/set-fn-validation! true))]}
+             :dev [:defaults
+                   {:dependencies [[org.bouncycastle/bcpkix-jdk15on]]}]
+             :fips [:defaults
+                    {:dependencies [[org.bouncycastle/bcpkix-fips]
+                                    [org.bouncycastle/bc-fips]
+                                    [org.bouncycastle/bctls-fips]]
+
+                     ;; this only ensures that we run with the proper profiles
+                     ;; during testing. This JVM opt will be set in the puppet module
+                     ;; that sets up the JVM classpaths during installation.
+                     :jvm-opts ~(let [version (System/getProperty "java.version")
+                                      [major minor _] (clojure.string/split version #"\.")
+                                      unsupported-ex (ex-info "Unsupported major Java version. Expects 8 or 11."
+                                                       {:major major
+                                                        :minor minor})]
+                                  (condp = (java.lang.Integer/parseInt major)
+                                    1 (if (= 8 (java.lang.Integer/parseInt minor))
+                                        ["-Djava.security.properties==dev-resources/jdk8-fips-security"]
+                                        (throw unsupported-ex))
+                                    11 ["-Djava.security.properties==dev-resources/jdk11-fips-security"]
```

Comment:
Do we need to package this security file in ezbake? Or will it ship with another package/be created by the PE module?

Reply:
no, that's handled by puppetlabs/bouncy-castle-vanagon#4 and puppetlabs/puppet-enterprise-modules#673